delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/RealmsDummy.scala

```diff
@@ -1,61 +1,41 @@
+import akka.persistence.query.Offset
 import cats.effect.Clock
-import cats.implicits._
 import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
 import ch.epfl.bluebrain.nexus.delta.sdk.model.realms.RealmCommand.{CreateRealm, DeprecateRealm, UpdateRealm}
-import ch.epfl.bluebrain.nexus.delta.sdk.model.realms.RealmRejection.{RevisionNotFound, UnexpectedInitialState}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.realms.RealmRejection.UnexpectedInitialState
 import ch.epfl.bluebrain.nexus.delta.sdk.model.realms.RealmState.Initial
 import ch.epfl.bluebrain.nexus.delta.sdk.model.realms._
 import ch.epfl.bluebrain.nexus.delta.sdk.model.search.Pagination.FromPagination
-import ch.epfl.bluebrain.nexus.delta.sdk.model.search.ResultEntry.UnscoredResultEntry
 import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchParams.RealmSearchParams
 import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchResults.UnscoredSearchResults
 import ch.epfl.bluebrain.nexus.delta.sdk.model.{Envelope, Label, Name}
 import ch.epfl.bluebrain.nexus.delta.sdk.testkit.RealmsDummy._
-import ch.epfl.bluebrain.nexus.delta.sdk.{RealmResource, Realms}
-import ch.epfl.bluebrain.nexus.testkit.{IORef, IOSemaphore}
-import fs2.Stream
+import ch.epfl.bluebrain.nexus.delta.sdk.{Lens, RealmResource, Realms}
 import monix.bio.{IO, Task, UIO}
 
 /**
   * A dummy Realms implementation that uses a synchronized in memory journal.
   *
   * @param resolveWellKnown get the well known configuration for an OIDC provider resolver
-  * @param journal          a ref to the journal containing all the events discriminated by label
-  * @param cache            a ref to the cache containing all the current realm resources
-  * @param semaphore        a semaphore for serializing write operations on the journal
-  * @param maxStreamSize    truncate event stream after this size
+  * @param journal          the journal to store events
+  * @param cache            the cache to store resources
   */
 final class RealmsDummy private (
-    resolveWellKnown: Uri => IO[RealmRejection, WellKnown],
-    journal: IORef[RealmsJournal],
-    cache: IORef[RealmsCache],
-    semaphore: IOSemaphore,
-    maxStreamSize: Long
+    journal: RealmsJournal,
+    cache: RealmsCache,
+    resolveWellKnown: Uri => IO[RealmRejection, WellKnown]
 )(implicit clock: Clock[UIO])
     extends Realms {
-
-  private val offsetMax = new AtomicLong()
-
-  private def journalMap: UIO[Map[Label, RealmsJournal]] =
-    journal.get.map { v =>
-      v.groupBy(_.event.label).map { case (k, v) =>
-        k -> v.sortBy(_.sequenceNr)
-      }
-    }
-
   override def create(
       label: Label,
       name: Name,
       openIdConfig: Uri,
       logo: Option[Uri]
   )(implicit caller: Subject): IO[RealmRejection, RealmResource] =
-    eval(CreateRealm(label, name, openIdConfig, logo, caller)).flatMap(setToCache)
+    eval(CreateRealm(label, name, openIdConfig, logo, caller)).flatMap(cache.setToCache)
```

Comment:
cache.setToCache can be moved to eval for comprehension (like for orgs, projects)